### PR TITLE
DEV2-2535: Use IntelliJ Proxy settings

### PR DESCRIPTION
**Pull request by assapir**
By default, we are getting the proxy settings from IntelliJ and inheriting them to the binary.
This won't happen if the checkbox is disabled.
Was able to see the binary requests via `mitmproxy` when the setting was on.

Process env var in windows:
image

windows proxy:
image

### CODE REVIEW

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.kt [hunk 1]
```diff
@@ -12,12 +18,40 @@ open class BinaryProcessGateway {
 
     @Throws(IOException::class)
     open fun init(command: List<String?>?) {
-        val createdProcess = ProcessBuilder(command).start()
+        val pb = ProcessBuilder(command)
```
Comment (yonip23) [inline src/main/java/com/tabnine/binary/BinaryProcessGateway.kt:21]: rename here and in the function argument to `processBuilder`

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.kt [hunk 2]
```diff
@@ -12,12 +18,40 @@ open class BinaryProcessGateway {
...
+    private fun setProxyEnvironmentVariables(pb: ProcessBuilder) {
+        val env = pb.environment()
+        val serverUrl = StaticConfig.getBundleServerUrl()
+        if (serverUrl.isPresent) {
+            val serverUrlString = serverUrl.get()
+            val tabnineServerProxy = CommonProxy.getInstance().select(URL(serverUrlString)).firstOrNull()
+            if (tabnineServerProxy?.type() == Proxy.Type.DIRECT) {
+                env["NO_PROXY"] = serverUrlString
+                env["no_proxy"] = serverUrlString
+            }
```
Comment (yonip23) [inline src/main/java/com/tabnine/binary/BinaryProcessGateway.kt:40]: extract to function `setOurServerUrlProxy`

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.kt [hunk 3]
```diff
@@ -12,12 +18,40 @@ open class BinaryProcessGateway {
...
+            return
+        }
+        val proxyString = "${HttpConfigurable.getInstance().PROXY_HOST}:${HttpConfigurable.getInstance().PROXY_PORT}"
+
+        if (proxyString.isNotBlank()) {
+            env["HTTPS_PROXY"] = proxyString
+            env["https_proxy"] = proxyString
+            env["HTTP_PROXY"] = proxyString
+            env["http_proxy"] = proxyString
+        }
```
Comment (yonip23) [inline src/main/java/com/tabnine/binary/BinaryProcessGateway.kt:52]: extract to function `inheritIdeProxy`

File: src/main/java/com/tabnine/binary/BinaryProcessGateway.kt (hunk 2 above)
Comment (amirbilu) [reply src/main/java/com/tabnine/binary/BinaryProcessGateway.kt:40]: this should be the host, right? not including protocol

**Pull request merged**